modules/gapi/src/streaming/onevpl/engine/processing_engine_base.cpp

```diff
@@ -0,0 +1,130 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <opencv2/gapi/streaming/onevpl/data_provider_interface.hpp>
+#include "streaming/onevpl/engine/processing_engine_base.hpp"
+#include "streaming/onevpl/accelerators/accel_policy_interface.hpp"
+#include "logger.hpp"
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+ProcessingEngineBase::ProcessingEngineBase(std::unique_ptr<VPLAccelerationPolicy>&& accel) :
+    acceleration_policy(std::move(accel)) {
+}
+
+ProcessingEngineBase::~ProcessingEngineBase() {
+    GAPI_LOG_INFO(nullptr, "destroyed");
+}
+
+ProcessingEngineBase::ExecutionStatus ProcessingEngineBase::process(mfxSession session) {
+    auto sess_it = sessions.find(session);
+    if (sess_it == sessions.end()) {
+
+        // TODO remember the last session status
+        return ExecutionStatus::Processed;
```

Comment:
I don't get this logic, Why it is "processed"if it is not found?

Reply:
i used it to notify `bool pull(Data&)` about no-more-data-for-processing and not-an-error situation with `false` 
Actually, maybe new status is required `NoData` or something like... Currently we have only single session from Source and it is predictable situation.
Will add new status here & update UTs

Reply:
ok